PermissionScope.podspec

```diff
@@ -13,4 +13,91 @@ Pod::Spec.new do |s|
+    bluetooth.pod_target_xcconfig  = feature_flags
+    bluetooth.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'Location' do |location|
+    location.dependency 'PermissionScope/Core'
+    location.weak_framework       = 'CoreLocation'
+    feature_flags                 = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestLocationEnabled' }
+    location.pod_target_xcconfig  = feature_flags
+    location.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'Microphone' do |mic|
+    mic.dependency 'PermissionScope/Core'
+    mic.weak_framework       = 'AVFoundation'
+    feature_flags            = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestMicrophoneEnabled' }
+    mic.pod_target_xcconfig  = feature_flags
+    mic.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'PhotoLibrary' do |photo|
+    photo.dependency 'PermissionScope/Core'
+    photo.weak_framework       = 'Photos', 'AssetsLibrary'
+    feature_flags              = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestPhotoLibraryEnabled' }
+    photo.pod_target_xcconfig  = feature_flags
+    photo.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'Camera' do |cam|
+    cam.dependency 'PermissionScope/Core'
+    cam.weak_framework       = 'AVFoundation'
+    feature_flags            = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestCameraEnabled' }
+    cam.pod_target_xcconfig  = feature_flags
+    cam.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'Notifications' do |note|
+    note.dependency 'PermissionScope/Core'
+    note.weak_framework       = 'UIKit', 'UserNotifications'
+    feature_flags             = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestNotificationsEnabled' }
+    note.pod_target_xcconfig  = feature_flags
+    note.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'Contacts' do |contacts|
+    contacts.dependency 'PermissionScope/Core'
+    contacts.weak_framework       = 'Contacts', 'AddressBook'
+    feature_flags                 = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestContactsEnabled' }
+    contacts.pod_target_xcconfig  = feature_flags
+    contacts.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'Events' do |cal|
+    cal.dependency 'PermissionScope/Core'
+    cal.weak_framework       = 'EventKit'
+    feature_flags            = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestEventsEnabled' }
+    cal.pod_target_xcconfig  = feature_flags
+    cal.user_target_xcconfig = feature_flags
+  end
+
+  s.subspec 'Reminders' do |rem|
+    rem.dependency 'PermissionScope/Core'
+    rem.weak_framework       = 'EventKit'
+    feature_flags            = { 'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'PermissionScopeRequestRemindersEnabled' }
+    rem.pod_target_xcconfig  = feature_flags
+    rem.user_target_xcconfig = feature_flags
+  end
+
```

Comment:
Cool setup. :+1:

Have we tested yet with mixing/matching the subspecs to make sure that the flags are combined correctly?

I wonder if there's a terser, DRY-er way to define the subspecs without duplicating the stuff they have in common, but maybe that'd be too fancy.